lib/bemxjst/match.js

```diff
@@ -30,7 +30,19 @@ MatchNested.prototype.exec = function exec(context) {
       return false;
   }
 
-  return val[this.keys[i]] === this.value;
+  var keyVal = val[this.keys[i]];
+  var thisVal = this.value;
+
+  if (
+      keyVal === undefined ||
+      keyVal === null ||
+      thisVal === undefined ||
+      thisVal === null
+  ) {
+    return false;
+  }
+
+  return val[this.keys[i]].toString() === this.value.toString();
```

Comment:
Btw, what you think about overloaded toString methods? Is it possible to pass it here?